crates/ika-core/src/dwallet_mpc/crytographic_computation/mpc_computations/reconfiguration.rs

```diff
@@ -72,31 +72,51 @@ impl ReconfigurationV2PartyPublicInputGenerator for ReconfigurationV2Secp256k1Pa
             extract_encryption_keys_from_committee(&upcoming_committee)?;
         match network_dkg_public_output {
             VersionedNetworkDkgOutput::V1(network_dkg_public_output) => {
-                let VersionedDecryptionKeyReconfigurationOutput::V2(
-                    latest_reconfiguration_public_output,
-                ) = latest_reconfiguration_public_output
-                else {
-                    return Err(DwalletMPCError::InternalError(
-                    "Reconfiguration to V2 only supports reconfiguration public output of version V2"
-                        .to_string(),
-                ));
-                };
-                let public_input: <ReconfigurationV2Secp256k1Party as Party>::PublicInput =
-                <twopc_mpc::decentralized_party::reconfiguration::Party as mpc::Party>::PublicInput::new_from_reconfiguration_output(
-                    &current_access_structure,
-                    upcoming_access_structure,
-                    current_encryption_keys_per_crt_prime_and_proofs.clone(),
-                    upcoming_encryption_keys_per_crt_prime_and_proofs.clone(),
-                    current_tangible_party_id_to_upcoming(current_committee, upcoming_committee)
-                        .clone(),
-                    bcs::from_bytes(&network_dkg_public_output)?,
-                    bcs::from_bytes(&latest_reconfiguration_public_output)?,
-                )
-                    .map_err(DwalletMPCError::from)?;
-
-                Ok(public_input)
+                match latest_reconfiguration_public_output {
+                    None => {
+                        let public_input: <ReconfigurationV2Secp256k1Party as Party>::PublicInput =
+                            <twopc_mpc::decentralized_party::reconfiguration::Party as mpc::Party>::PublicInput::new_from_dkg_output(
+                                &current_access_structure,
+                                upcoming_access_structure,
+                                current_encryption_keys_per_crt_prime_and_proofs.clone(),
+                                upcoming_encryption_keys_per_crt_prime_and_proofs.clone(),
+                                current_tangible_party_id_to_upcoming(current_committee, upcoming_committee)
+                                    .clone(),
+                                bcs::from_bytes(&network_dkg_public_output)?,
+                            )
+                                .map_err(DwalletMPCError::from)?;
+
+                        Ok(public_input)
+                    }
```

Comment:
No, this case is not supported.

We have the reconfiguration v1 to v2 protocol because there is already a network key that has v1 reconfiguration output, but no network key has v1 dkg output only.

And if a new key will be generated after the update, it will be with the v2 output.

So this options does not exist, and an error should be returned, and the session rejected.